tests/python_package_test/test_sklearn.py

```diff
@@ -1050,7 +1050,7 @@ def test_inf_handle():
     ncols = 10
     X = np.random.randn(nrows, ncols)
     y = np.random.randn(nrows) + np.full(nrows, 1e30)
```

Comment:
```suggestion
    y = np.random.randn(nrows) + np.full(nrows, np.inf)
```

Reply:
how about this? @StrikerRUS

Reply:
sklearn prohibits passing `inf` directly.
```
ValueError                                Traceback (most recent call last)
<ipython-input-3-df0688c08f40> in <module>
      2 params_fit = {'X': X, 'y': y, 'sample_weight': weight, 'eval_set': (X, y), 
      3            'callbacks': [lgb.early_stopping(5)]} 
----> 4 gbm = lgb.LGBMRegressor(**params).fit(**params_fit)
      5 np.testing.assert_allclose(gbm.evals_result_['training']['l2'], np.inf)

D:\Miniconda3\lib\site-packages\lightgbm\sklearn.py in fit(self, X, y, sample_weight, init_score, eval_set, eval_names, eval_sample_weight, eval_init_score, eval_metric, feature_name, categorical_feature, callbacks, init_model)
    935     ):
    936         """Docstring is inherited from the LGBMModel."""
--> 937         super().fit(
    938             X,
    939             y,

D:\Miniconda3\lib\site-packages\lightgbm\sklearn.py in fit(self, X, y, sample_weight, init_score, group, eval_set, eval_names, eval_sample_weight, eval_class_weight, eval_init_score, eval_group, eval_metric, feature_name, categorical_feature, callbacks, init_model)
    675 
    676         if not isinstance(X, (pd_DataFrame, dt_DataTable)):
--> 677             _X, _y = _LGBMCheckXY(X, y, accept_sparse=True, force_all_finite=False, ensure_min_samples=2)
    678             if sample_weight is not None:
    679                 sample_weight = _LGBMCheckSampleWeight(sample_weight, _X)

D:\Miniconda3\lib\site-packages\sklearn\utils\validation.py in check_X_y(X, y, accept_sparse, accept_large_sparse, dtype, order, copy, force_all_finite, ensure_2d, allow_nd, multi_output, ensure_min_samples, ensure_min_features, y_numeric, estimator)
   1074     )
   1075 
-> 1076     y = _check_y(y, multi_output=multi_output, y_numeric=y_numeric, estimator=estimator)
   1077 
   1078     check_consistent_length(X, y)

D:\Miniconda3\lib\site-packages\sklearn\utils\validation.py in _check_y(y, multi_output, y_numeric, estimator)
   1096         estimator_name = _check_estimator_name(estimator)
   1097         y = column_or_1d(y, warn=True)
-> 1098         _assert_all_finite(y, input_name="y", estimator_name=estimator_name)
   1099         _ensure_no_complex_data(y)
   1100     if y_numeric and y.dtype.kind == "O":

D:\Miniconda3\lib\site-packages\sklearn\utils\validation.py in _assert_all_finite(X, allow_nan, msg_dtype, estimator_name, input_name)
    140                     " [https://scikit-learn.org/stable/modules/impute.html"](https://scikit-learn.org/stable/modules/impute.html%22)
    141                 )
--> 142             raise ValueError(msg_err)
    143 
    144     # for object dtype data, we only check for NaNs (GH-13254)

ValueError: Input y contains infinity or a value too large for dtype('float64').
```

But even if switch to standard API without weights, prediction result is `0.0` not `inf`, which I guess is a result of inner replacement of incorrect values with 0.0 #4483.
https://github.com/microsoft/LightGBM/blob/d517ba12f2e7862ac533908304dddbd770655d2b/src/io/metadata.cpp#L319

Also, I think the idea of the test was to get inf by multiplication not directly handle inf

Reply:
Hmmm, also, label is expected to be float32, not float64.
https://github.com/microsoft/LightGBM/blob/9627001c707aee1a650b911cd123d04280b580a8/python-package/lightgbm/basic.py#L455-L459

```python
import numpy as np
import lightgbm as lgb

nrows = 100
ncols = 10
X = np.random.randn(nrows, ncols)
y = np.random.randn(nrows) + np.full(nrows, 1e300)

lgb_data = lgb.Dataset(X, y).construct()
lgb_data.get_label()
```

Outputs

```
array([1.e+38, 1.e+38, 1.e+38, 1.e+38, 1.e+38, 1.e+38, 1.e+38, 1.e+38,
       1.e+38, 1.e+38, 1.e+38, 1.e+38, 1.e+38, 1.e+38, 1.e+38, 1.e+38,
       1.e+38, 1.e+38, 1.e+38, 1.e+38, 1.e+38, 1.e+38, 1.e+38, 1.e+38,
       1.e+38, 1.e+38, 1.e+38, 1.e+38, 1.e+38, 1.e+38, 1.e+38, 1.e+38,
       1.e+38, 1.e+38, 1.e+38, 1.e+38, 1.e+38, 1.e+38, 1.e+38, 1.e+38,
       1.e+38, 1.e+38, 1.e+38, 1.e+38, 1.e+38, 1.e+38, 1.e+38, 1.e+38,
       1.e+38, 1.e+38, 1.e+38, 1.e+38, 1.e+38, 1.e+38, 1.e+38, 1.e+38,
       1.e+38, 1.e+38, 1.e+38, 1.e+38, 1.e+38, 1.e+38, 1.e+38, 1.e+38,
       1.e+38, 1.e+38, 1.e+38, 1.e+38, 1.e+38, 1.e+38, 1.e+38, 1.e+38,
       1.e+38, 1.e+38, 1.e+38, 1.e+38, 1.e+38, 1.e+38, 1.e+38, 1.e+38,
       1.e+38, 1.e+38, 1.e+38, 1.e+38, 1.e+38, 1.e+38, 1.e+38, 1.e+38,
       1.e+38, 1.e+38, 1.e+38, 1.e+38, 1.e+38, 1.e+38, 1.e+38, 1.e+38,
       1.e+38, 1.e+38, 1.e+38, 1.e+38], dtype=float32)
```

Reply:
I think the root cause is that we set most float values to [-1e38, 1e38].  https://github.com/microsoft/LightGBM/blob/17d4e007852b4f0b34b211a674cb26e6751c98e2/src/io/metadata.cpp#L352
and the type of weight is also float32, not double.
therefore, the inf will not produce. 
In this case, I think we can remove the test.

Reply:
> I think we can remove the test.

Done in 6aa9d11087ef0a27734f929d6625e5a822cabc2c.